### PR TITLE
[iOS] Added Google Maps Circle, Polygon, Polyline, MapType Support

### PR DESCRIPTION
**Pull request by unboundfire**
Needed the ability to draw on maps and switch the base map using the Google Maps SDK so I took a stab at adding this support.

### CODE REVIEW

**Comment by volkangurel** [comment]
👍

File: components/MapView.js [hunk 1]
```diff
@@ -502,9 +498,6 @@ class MapView extends React.Component {
         onMapReady: this._onMapReady,
         onLayout: this._onLayout,
       };
-      if (Platform.OS === 'ios' && ANDROID_ONLY_MAP_TYPES.includes(props.mapType)) {
-        props.mapType = MAP_TYPES.STANDARD;
-      }
```
Comment (gilbox) [inline components/MapView.js]: did you consider keeping this check in case someone is still using mapkit?

Comment (unboundfire) [inline components/MapView.js]: @gilbox good point! I can add this check back and base it off the platform and provider

**Comment by unboundfire** [comment]
Added the check back

**Comment by gilbox** [comment]
this lgtm.. @spikebrehm wanna take a peek?

**Comment by unboundfire** [comment]
@gilbox @spikebrehm Just added circle support

**Comment by enagy27** [comment]
👍

**Comment by Stonelinks** [comment]
:+1:

**Comment by manelpb** [comment]
This is perfect! Guys please merge this pull-request!!

**Comment by spikebrehm** [comment]
LGTM! Thanks 🍻

**Pull request merged**